### DISCUSSION TITLE
How to plot multiple elements without overlapping automatically

### DISCUSSION THREAD

**Discussion by BruceChen2017** [clientIO/joint#1871] [Q&A]
Recently , I am playing the demo of [chatbot](https://resources.jointjs.com/demos/chatbot).  From backend, we will receive multiple entities each having label and description. In this case, how can plot all these elements **_programmatically_** such that all the elements will not _overlap_ and have a reasonable layout? 

P.S. Actually I also don't how to create an entity _programmatically_ now from chatbot

**Comment by kumilingus** [comment]
You are looking for an automatic layout. It depends what kind of diagram you have (what are the relationships between elements).

### JointJS
- [Directed Graph Layout](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#layout.DirectedGraph) 
- [Circular Layout](https://codepen.io/jointjs/pen/QWxjjdz)
- [Rectangular Layout](https://codepen.io/jointjs/pen/zYaxzxB)
- This [demo](https://resources.jointjs.com/demos/elk) shows how to use the very powerful [Eclipse Kernel Layout](https://github.com/kieler/elkjs).

### JointJS+
- [Tree Layout](https://resources.jointjs.com/docs/rappid/v3.6/layout.html#layout.TreeLayout)
- [Grid Layout](https://resources.jointjs.com/docs/rappid/v3.6/layout.html#layout.GridLayout)
- [Force Directed Layout](https://resources.jointjs.com/docs/rappid/v3.6/layout.html#layout.ForceDirectedLayout)

**Reply by BruceChen2017** [reply]
https://resources.jointjs.com/docs/rappid/v3.6/layout.html#layout.TreeLayout is broken

**Reply by jamesgeorgewilliams** [reply]
Hi, those are the old URLs, sorry. You can find the updated ones here:

https://resources.jointjs.com/docs/rappid/v3.6/layout.TreeLayout.html
https://resources.jointjs.com/docs/rappid/v3.6/layout.GridLayout.html
https://resources.jointjs.com/docs/rappid/v3.6/layout.ForceDirected.html
